app/src/main/java/com/example/farmeraid/data/FarmRepository.kt

```diff
@@ -64,5 +61,22 @@ class FarmRepository(
 
     }
 
+    suspend fun getMarketIds (): ResponseModel.FAResponseWithData.Success<MutableList<String>> {
+        Log.d("TEST", "called")
+        return (userRepository.getFarmId()?.let { id ->
+            try {
+                db.collection("farm").document(id)
+                    .get()
+                    .await()
+                    .data?.get("markets")?.let {
+                        ResponseModel.FAResponseWithData.Success(it as MutableList<String>)
+                    } ?: ResponseModel.FAResponseWithData.Error("Error fetching markets")
+            } catch (e : Exception) {
+                Log.e("InventoryRepository", e.message ?: e.stackTraceToString())
+                ResponseModel.FAResponseWithData.Error(e.message ?: "Unknown error while getting inventory")
+            }
+        } ?: ResponseModel.FAResponseWithData.Error("User is not part of a farm")) as ResponseModel.FAResponseWithData.Success<MutableList<String>>
```

Comment:
Once you make the previous change, you can remove the `as ResponseModel.FAResponseWithData.Success<MutableList<String>>`